greenpeace/planet4-master-theme
# PLANET-7564: Implement media replacement feature

Ref: https://jira.greenpeace.org/browse/PLANET-7564

<hr>

**DESCRIPTION:** 
This PR replaces all kinds of files (except images) from the Media Library both in the local `uploads` folder and Google Storage while keeping the original files' names. It also purges Cloudflare's cache once the file is replaced.

<hr>

**TESTING:**

- **Proteus environment:**
  - Go to the Media Library page: `/wp-admin/upload.php`
  - Select the file to replace (or add a new one if all the files in the library are images).
  - Click on the Replace Media button.
  - Wait for the page to reload.
  - Check that the file was replaced in the [Google Storage bucket](https://console.cloud.google.com/storage/browser/planet4-test-proteus-stateless-develop).
- **Local environment:**
  - _Local replacements:_
    - Go to the Media Library page: `/wp-admin/upload.php`
    - Select the file to replace (or add a new one if all the files in the library are images).
    - Click on the Replace Media button.
    - Wait for the page to reload.
    - Check that the file in the `planet4/uploads` folder was replaced.
  - _Google Storage replacements:_
    - Copy the [WP Stateless configuration](https://www-dev.greenpeace.org/gutenberg/wp-admin/upload.php?page=stateless-settings) of the Gutenberg environment
    - Go to the Media Library page: `/wp-admin/upload.php`
    - Select the file to replace (or add a new one if all the files in the library are images).
    - Click on the Replace Media button.
    - Wait for the page to reload.
    - Check that the file was replaced in the [Gutenberg's Google Storage bucket](https://console.cloud.google.com/storage/browser/planet4-test-gutenberg-stateless-develop).
